rcloud.support/R/assets.R

```diff
@@ -63,12 +63,18 @@ rcloud.upload.asset <- function(name, content, notebook=rcloud.session.notebook(
         if (f$isdir) stop("cannot upload a directory")
         tryCatch(content <- readBin(file, raw(), f$size), warning=function(e) stop(e$message))
     }
-    if (is.list(notebook))
+
+    if (is.list(notebook)){
         notebook <- notebook$content$id
+    }
     l <- list(list(content=content))
     names(l) <- name
     invisible(rcloud.update.notebook(notebook, list(files=l)))
 }
 
-rcloud.delete.asset <- function(name, notebook=rcloud.session.notebook())
-    rcloud.upload.asset(name, NULL, notebook)
+rcloud.delete.asset <- function(name, notebook=rcloud.session.notebook()) {
+    if (name=="thumb.png"){
+        rcloud.set.thumb(NULL, notebook$content$id)
```

Comment:
Asset deletion doesn't usually pass through here; I think this is an API for client code. Usually we'll send a batch of changes, including `filename: null`; I provided a link to the GitHub API in a comment.